src/Couchbase.Lite.Shared/Sync/ReplicatorOptionsDictionary.cs

```diff
@@ -195,6 +215,22 @@ public bool Reset
             }
         }
 
+        /// <summary>
+        /// Gets or sets the replicator heartbeat keep-alive interval.
+        /// </summary>
+        public TimeSpan Heartbeat
+        {
+            get => TimeSpan.FromSeconds(this.GetCast<long>(HeartbeatIntervalKey));
+            set { 
+                var sec = value.TotalSeconds;
+                if (sec > 0) {
```

Comment:
Be careful, `TotalSeconds` is a double type.  You should floor it first or 0.5, for example, will turn into 0 later and give unexpected results.

Reply:
Change to value.Ticks/TimeSpan.TicksPerSecond. So when user assign TimeSpan.FromMilliseconds(800) to heartbeat interval, the invalid exception will throw.

Reply:
I know the error message is meant to be cross platform but it will be confusing that the user gets "Heartbeat Interval cannot be less or equal to 0."  Consider "Heartbeat Interval cannot be less or equal to 0 full seconds.